calculatorBackend/src/Rest/inc/Rest.h

```diff
@@ -0,0 +1,40 @@
+#ifndef REST_H
+#define REST_H
+
+////////////////////////////////////////////////////////////////////////////////
+/**
+* @file Rest.h
+* @brief rest function declarations.
+*/
+////////////////////////////////////////////////////////////////////////////////
+
+#include "pistache/endpoint.h"
+#include "pistache/router.h"
+
+namespace calculator {
+
+/**
+* @brief Rest interface which exposes to other components
+*/
+
+class Rest
+{
+    public:
+        /**
+         * @brief handler for when client hits the calculate route
+         * @param [in] request the pistache data needed to handle server requests
+         * @param [in] response the pistache data needed to handle the response from the server
+        */
+        void calculateRoute(const Pistache::Rest::Request& request, Pistache::Http::ResponseWriter response);
```

Comment:
I don't like the function names here. I asked Gary his thoughts and had a good discussion on a better naming scheme. Feel free to implement this if you like.

Rename the class(and filename) to CalculateProvider; since it provides the implementation on the calculate endpoint. Then rename calculateRoute to `post` and optionsTest to `options` so it directly maps to the http requests that they are servicing.

Let me know if you have any questions.

The idea would be is if you wanted to add another endpoint (/history) you could create another class called HistoryProvider and structure it similarly.